### PR TITLE
Add basic testing page for browser quicks

### PR DESCRIPTION
**Pull request by nhunzaker**
Related to #8583

---

This is just the start. I'd like to dynamically pull in versions some how, and @gaearon mentioned renaming `examples` to `fixtures`, but it accomplishes most of what we need (I think).

Next step would be to create a basic template for a fixture.

test-page

---

@Aweary Should we start working from a branch other than master? I can't do that, but I'd be happy to adjust the base of this PR to a `browser-quirks` (or whatever) branch on the main repo.

### CODE REVIEW

File: examples/basic/index.html [hunk 1]
```diff
@@ -23,8 +23,9 @@
       Learn more about React at
       <a href="https://facebook.github.io/react" target="_blank">facebook.github.io/react</a>.
     </p>
-    <script src="../../build/react.js"></script>
-    <script src="../../build/react-dom.js"></script>
+
+    <script src="../react-loader.js"></script>
```
Comment (nhunzaker) [inline examples/basic/index.html]: I've only done it for the basic example, but I could imagine doing this everywhere. We could also add an extra flag to pull in Babel.

File: examples/test.html [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          <select id="react_version" name="version">
+            <option value="local">Local</option>
+            <option value="15.4.1">15.4.1</option>
+            <option value="15.4.0">15.4.0</option>
+            <option value="15.3.2">15.3.2</option>
+            <option value="15.3.1">15.3.1</option>
+            <option value="15.3.0">15.3.0</option>
+            <option value="15.2.1">15.2.1</option>
+            <option value="15.1.0">15.1.0</option>
+          </select>
```
Comment (nhunzaker) [inline examples/test.html]: Totally arbitrary and temporary list. What's a good way to pull in all of the version data?

**Comment by Aweary** [comment]
> @Aweary Should we start working from a branch other than master? I can't do that, but I'd be happy to adjust the base of this PR to a browser-quirks (or whatever) branch on the main repo.

I think I can just push changes to your branch, as long as you didn't uncheck that option when opening the PR.

File: examples/test.html (hunk 2 above)
Comment (Aweary) [inline examples/test.html]: It would be nice if unpkg or npm had a small API to get this information, but it doesn't look like they do. The easiest way may be to just manually keep it up to date as part of the release process. It would be nice to have something like CHANGELOG but in JSON.

**Comment by Aweary** [comment]
This is a good start 👍 How are we going to structure things at a high-level? Should tests be grouped by together in separate pages (`input.html`, `textarea.html`, `select.html`, etc.) or are we going to just keep it on a single page?

Personally, I think separate pages is the way to go since I expect this to get pretty large by the time we're done.

If we *do* use separate pages, does the `iframe` approach make sense? Would toggling between different pages work like toggling between different React versions?

**Comment by nhunzaker** [comment]
> I think I can just push changes to your branch, as long as you didn't uncheck that option when opening the PR.

Woah neat. Good deal.

I think we should do separate pages too. We could add a dropdown just like React versions. Maybe setup a script that generates it (or do it by hand for a while).

Another part of this is documentation. Where does documentation for fixes to random browser issues live?

File: examples/test.html (hunk 2 above)
Comment (vinnymac) [inline examples/test.html]: @nhunzaker If you make a request to the NPM registry and parse out the version keys, you should be able to get a similar output to the `npm view react versions` command. Here is the the response on the registry for [react](https://registry.npmjs.org/react), it contains all versions published. It may be more information than you need, but I hope it helps. (Note: You could fall back to [Yarn](https://registry.yarnpkg.com/react) if NPM was down.)

Comment (nhunzaker) [inline examples/test.html]: @vinnymac Very cool. I think tags is precisely what I want.

Comment (nhunzaker) [inline examples/test.html]: Added. We'll pull from the Github API, though this fails in IE9 because of silliness with CORS and XDomainRequest. Something to figure out later... 

Otherwise, this page looks good and works down to IE9.

**Comment by nhunzaker** [comment]
Updated the readme in `examples/inputs` to talk about the difference between `defaultValue` and `value`, and the concept of value detachment.

**Comment by nhunzaker** [comment]
@gaearon mentioned renaming `examples` to `fixtures`. For the sake of keeping this up to date with master, I'll go ahead and make a fixtures folder and move all of this work over to that. We can always rename/remove examples at a later point.

File: fixtures/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+Each document should be browsable without setting up a static file server. Simply
+open the test case in a browser.
+
+## Browser Targets
+
+IE9+
+IE Edge
+Chrome
+Firefox
+Safari
```
Comment (nhunzaker) [inline fixtures/README.md]: Curiously, I couldn't find anything about base line browser support for React. Would it be the same as Facebook? Either way, do you know what this support matrix looks like?

Otherwise, drawing from the list of ES5 compliant browsers makes the most sense to me:
http://caniuse.com/#search=ES5

Comment (nhunzaker) [inline fixtures/README.md]: Ah... neat. Thank you!

**Comment by gaearon** [comment]
Excited to see work on this. Let me know if you get blocked on something!

**Comment by nhunzaker** [comment]
@gaearon you're fast! I was just writing this up..

Added:

- Test for selects
- Test for range inputs
- Test for textareas

I also write up a section on `form.reset()`, documented the current issues with number inputs, and moved all of this work to a `fixtures` folder.

At this point, I feel good about everything. The only outstanding issues relates to fetching the version information in IE9 for the dropdown menu. 

I think a next step would be to catalog past issues. I can think of two off the top of my head:

1. Disabled buttons/inputs
2. Assignment of min/max/step before value (causing display value issues in IE)
3. Display value issues with improperly detached date inputs in iOS Safar /Android Chrome

Though I'm sure there are more. 

Still, I think these are additive. At some point, I would love to check this in so that others can contribute to it more easily.

**Comment by nhunzaker** [comment]
And... just fixed the lint issues. Hopefully CI Passes now.

**Comment by aweary** [comment]
@nhunzaker I added a small commit to show the values for the controlled `<textarea/>` and `<select/>`. I think we should make sure to always display the output for controlled components.

I think we need to decide on some high-level design structure before adding anything else. We want to have an accessible set of test cases that anyone can run through. It should be easy enough for a new contributor to intuitively follow. IMO the clearest way to do that is to have instructions inline.
Then we'd have a page with a set of generalized components (controlled, uncontrolled, etc.) and then below them a set of interactive test cases. That would look something like:

screen shot 2016-12-22 at 9 38 50 am


The advantages with this approach is that it makes it very, very easy for a contributor to open this page and work from a central place to test their changes. We could even have little checkboxes that people can use to track which tests they've done every time their refresh/check changes.

The downside is that our component structure increases in complexity. Each individual test case would likely need to maintain its own state and all that. If we did it this way, we'd likely want to start thinking about a better structure (instead of a bunch of inlined components)

Alternatively we could have them listed in each individual README. The advantage there is that its probably easier to maintain and easier to read over. The downside is that it makes it slightly harder to track what you're doing and requires you to have two pages open at once.

What do you think?

**Comment by nhunzaker** [comment]
I like the first approach. Especially if we can make it available online. Each file would be a stand alone issue reproducer, making it easier to produce and avoids possibly missing context.

So let's start making cases?

**Comment by aweary** [comment]
Should we move away from loading static HTML files and provide a server that users can run? It would make future deployment efforts easier. Then we could actually separate stuff out into different files and organize a bit better. And I think we'd sidestep CORS issue too since we can just request version history from the server. 

Then users can just do `npm run fixtures` or something

**Comment by gaearon** [comment]
I feel like there's a lot of bikeshed potential if you move into the directions of servers and many pages. I would probably go with a single page and many components. Or you can use existing infra like Create React App.

**Comment by aweary** [comment]
I'm not saying do many individual pages, the current strategy of toggling between the different categories of components with a dropdown is a good approach. I'm just saying we can serve this file from `localhost` instead of just from `file://`. It would just be a simple webpack setup serving essentially the same thing we have now, except we wouldn't have to inline everything in a `script` tag.

> Or you can use existing infra like Create React App.

I like that idea! That actually seems like the most logical way to do it.

**Comment by nhunzaker** [comment]
>  It would just be a simple webpack setup serving essentially the same thing we have now

Maybe we could even.... get incremental react builds!

**Comment by nhunzaker** [comment]
I'm fine with either `file://` vs `http://`, whatever is the easiest to author and set up for new contributors. 

As for writing the tests themselves:

What I like about @aweary's decimal dropping example is that it includes specific reproduction steps for a confusing issue. Does it make sense to have a few pages for those strange edge cases, however maintain large pages by component type like we have now?

I can see a few focused examples from my past experience:

1. The `<input type=range />` hit on IE
2. Disabled button bubbling
3. Chrome/Safari backspace issues
4. Chrome/Safari DateTime value detachment issues

So pages for a particular problem. Ones that require special handling inside of React, or we don't want to lose history on.

Does this make sense?

**Comment by nhunzaker** [comment]
Also, I'm fine punting on specific test cases in favor of the "kitchen sink" style pages until we ship this. To me, they are more of documentation than anything else. 

@aweary What does your availability look like? Would you like to own setting up a `fixtures` `create-react-app`? Otherwise I can probably get to it either tonight, or next Tuesday.

**Comment by aweary** [comment]
@aweary I can own getting a fixtures app setup using `create-react-app` 🤘

**Comment by aweary** [comment]
> Does it make sense to have a few pages for those strange edge cases, however maintain large pages by component type like we have now?

I think we should have large pages for each category of components like now, and just have the general "kitchen sink" playground at the top, and then have a section below that lists each test case. That way if we provide a checkbox for each one they can just run through the list quickly and verify everything works with their changes, and it's easy to just scroll the page and see if there are any cases you didn't hit.

So it would go:

# Kitchen Sink / Playground
________

**Comment by aweary** [comment]
So I looked into getting an app setup using `create-react-app` and it's unfortunately not as straightforward as I'd hoped. Since we need to dynamically load different versions of `React` and `ReactDOM`, we need to make sure webpack doesn't bundle those. We can do that my marking them as external and making sure `React` and `ReactDOM` are available  *before* the bundle is loaded, but that means we need to eject and I think that would defeat the purpose of using `create-react-app` in the first place (we now have our own infra to maintain anyways).

**Comment by gaearon** [comment]
You can just read them from the global. CRA doesn't force you to import React in a specific way.

```
const ReactDOM = window.ReactDOM;
const React = window.React;
const Page = createPage(React, ReactDOM);
ReactDOM.render(<Page />, el);
```

**Comment by aweary** [comment]
I somehow didn't even think of doing that, thanks!

**Comment by aweary** [comment]
@nhunzaker @gaearon I've pushed my commit with the application created using `create-react-app` (https://github.com/facebook/react/pull/8589/commits/99bfa2892b1c2843d6a23b520f735925967d1a46) . It works pretty well overall. There are some notes in the commits about implementation details, and I'm not 100% on the folder structure, but I wanted to get something up for review 👍

File: fixtures/README.md [hunk 4]
```diff
@@ -1,16 +1 @@
...
-## Browser Targets
-
-React 15+ supports all ES5 compliant browsers:
-
-https://facebook.github.io/react/docs/react-dom.html#browser-support
-
-### IE8 Support
-
-React 0.14 and lower support IE8 after including several required shims.
+# React Fixtures (TODO)
```
Comment (aweary) [inline fixtures/README.md]: This got deleted when I merged, I'll make sure to add it back

File: fixtures/package.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "_fixtures",
```
Comment (aweary) [inline fixtures/package.json]: What should we call it? `react-fixtures`?

File: fixtures/public/react-loader.js [hunk 6]
```diff
@@ -7,8 +7,8 @@
  * (Loads React 15.4.1)
  */
 
-var REACT_PATH = '../../build/react.js';
-var DOM_PATH = '../../build/react-dom.js';
+var REACT_PATH = 'react.js';
+var DOM_PATH = 'react-dom.js';
```
Comment (aweary) [inline fixtures/public/react-loader.js]: We need to figure out the best way to get a copy of the local `react` and `react-dom` builds.  The relative path was not working out of the box, so I ended up just copying the files from `build` manually. Maybe we should make that part of the build process, or do it before the `start` script for this app?

If you're testing locally, make sure to copy those files into here for now!

File: fixtures/src/components/Header.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+const Header = React.createClass({
+  getInitialState() {
+    const query = parse(window.location.search);
+    const version = query.version || 'local';
+    const versions = [version];
+    return { version, versions };
+  },
+  componentWillMount() {
+    fetch('http://api.github.com/repos/facebook/react/tags')
```
Comment (aweary) [inline fixtures/src/components/Header.js]: Are we all OK with shipping this with a `fetch` polyfill so we can use this ^?

**Comment by nhunzaker** [comment]
Wow. Time to catch up! Happy new year!

File: fixtures/src/components/Header.js (hunk 7 above)
Comment (nhunzaker) [inline fixtures/src/components/Header.js]: Fine by me. `create-react-app` comes with it already too.

File: fixtures/public/react-loader.js (hunk 6 above)
Comment (nhunzaker) [inline fixtures/public/react-loader.js]: A `prestart` command that copies everything over is fine by me.

File: fixtures/public/react-loader.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+ * Take a version from the window query string and load a specific
+ * version of React.
+ *
+ * @example
+ * http://localhost:3000?version=15.4.1
+ * (Loads React 15.4.1)
+ */
+
+var REACT_PATH = 'react.js';
+var DOM_PATH = 'react-dom.js';
```
Comment (aweary) [inline fixtures/public/react-loader.js]: Question got hidden as "outdated", but its not: https://github.com/facebook/react/pull/8589#discussion_r94507090

File: fixtures/package.json (hunk 5 above)
Comment (nhunzaker) [inline fixtures/package.json]: 👍.

**Comment by aweary** [comment]
@nhunzaker :tada: everything is mostly the same as far as the actual fixtures go, I just moved them into their own files and got linting to pass.

File: fixtures/src/components/Header.js [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+      <div className="header__inner">
+        <span className="header__logo">
+          <img src="https://facebook.github.io/react/img/logo.svg" alt="" width="32" height="32" />
+          React Sandbox (v{React.version})
+        </span>
+
+        <div className="header-controls">
+          <label htmlFor="example">
+            <span className="sr-only">Select an example</span>
+            <select value={window.location.pathname} onChange={this.handleFixtureChange}>
```
Comment (nhunzaker) [inline fixtures/src/components/Header.js]: Should this value be in initial state?

File: fixtures/src/components/fixtures/range-inputs/index.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  },
+  onChange(event) {
+    this.setState({ value: event.target.value });
+  },
+  render() {
+    return (
+      <form>
+        <fieldset>
+          <legend>Controlled</legend>
+          <input type="range" value={this.state.value} onChange={this.onChange.bind(this)} />
```
Comment (nhunzaker) [inline fixtures/src/components/fixtures/range-inputs/index.js]: no need to bind `onChange` if we're using `createClass`

Comment (nhunzaker) [inline fixtures/src/components/fixtures/range-inputs/index.js]: Probably my mistake :)

**Comment by nhunzaker** [comment]
This looks great! in #8583 I had two other items:

- [] Catalog existing fixes and outstanding issues, start building test cases
- [] Modify React flags via query strings, or some other mechanism to quickly switch between the Fiber and and standard DOM renderer.

Would you consider the first to be reasonably complete, and do we want to cover feature flags in this PR?

File: fixtures/src/components/Header.js (hunk 9 above)
Comment (aweary) [inline fixtures/src/components/Header.js]: Yeah, probably. The way I do it with version is to just pass it from the `window.location` and update `window.location.search` when it changes, forcing it to reload (since we need a full reload on version change).

I did the same thing here, but I guess we don't need a full reload on fixture change?

**Comment by nhunzaker** [comment]
Naw, we don't need to reload, but I don't know how much we want to mess
with history API support. Worth it, you think?

File: fixtures/src/components/Header.js (hunk 9 above)
Comment (nhunzaker) [inline fixtures/src/components/Header.js]: Reply via email didn't match up:

> Naw, we don't need to reload, but I don't know how much we want to mess
with history API support. Worth it, you think?

**Comment by nhunzaker** [comment]
Hey, fun! Looks like we're hitting the new `document.write` thing in Chrome:

screen shot 2017-01-03 at 7 10 22 pm

Joy! I'll fix this.

**Comment by nhunzaker** [comment]
Well.. we don't need babel for sure anymore. I think the other thing was related to a local build. I'll add that pre-start script.

**Comment by nhunzaker** [comment]
@aweary That detail work, yo:

screen shot 2017-01-03 at 7 17 53 pm

🔥

**Comment by nhunzaker** [comment]
Pushed some stuff:

- remove babel include from react-loader
- Copy local build before starting
- Add test prompt message when no fixture is selected
- Remove extraneous binds from some fixtures

File: fixtures/src/components/Header.js (hunk 9 above)
Comment (aweary) [inline fixtures/src/components/Header.js]: Yeah, I already feel unsure about setting `window.location` properties directly, does that consistently cause a refresh in all supported browsers? If we don't need to do it then 👍 on avoid it

Comment (nhunzaker) [inline fixtures/src/components/Header.js]: It sure does.

Comment (nhunzaker) [inline fixtures/src/components/Header.js]: Oh, herm... I'll check. Be back shortly.

**Comment by gaearon** [comment]
What is the status here?

**Comment by nhunzaker** [comment]
@gaearon doing some browser testing. I'll report back shortly.

**Comment by aweary** [comment]
> What is the status here?

We've setup an application using `create-react-app` (see https://github.com/facebook/react/pull/8589/commits/99bfa2892b1c2843d6a23b520f735925967d1a46 and the handful of commits @nhunzaker has made since) with the ~same content we had before. Just wanting to see if you had any feedback on the overall app structure + implementation. No rush 👍 

> Would you consider the first to be reasonably complete, and do we want to cover feature flags in this PR?

@nhunzaker I think we can implement feature flags in this PR and continue documenting test cases in #8583. It might be good to merge this soon and then add the other test cases in future PRs, so this doesn't get too big.

**Comment by gaearon** [comment]
Can we move descriptions of test cases inline inside the page? So that you can just fire it up and walk through the whole thing without switching applications.

**Comment by aweary** [comment]
@gaearon currently this PR just contains the general kitchen sink components. We still need to create the individual test cases.

I'm thinking we can have a component like `<FixtureTestCase />` that lets you define a set of steps and provides an interface for marking the case as "tested" so you can track progress as you go through the page.

```jsx
<FixtureTestCase title="Foo" steps={steps}>
  // components for test case go here
</FixtureTestCase>

**Comment by gaearon** [comment]
>We still need to create the individual test cases, if that's what you mean?

Yea.

File: fixtures/src/components/Header.js (hunk 9 above)
Comment (nhunzaker) [inline fixtures/src/components/Header.js]: Checks out for me down to IE9 and all major browsers

**Comment by nhunzaker** [comment]
Did some cross browser checking/fixing. Also simplified styles into a single file, we needed the input form styles in the textarea example too. 

So the outstanding tasks:

- [ ] Individual test cases
- [ ] Move descriptions of test cases inline inside the page? 

@gaearon would you say that this is correct?

**Comment by nhunzaker** [comment]
Fixed lint issues. Though I think I'm hitting this locally:

```
Cannot find module 'eslint-config-postcss/es5'
Referenced from: <...>/react/fixtures/node_modules/autoprefixer/package.json
Error: Cannot find module 'eslint-config-postcss/es5'
```

So let's see what CI says. Curious why this isn't throwing on CI though.

**Comment by gaearon** [comment]
I just merged a folder named `fixtures` in an unrelated commit (for manual testing of build systems). Could you please move its content into `fixtures/build` and make this `fixtures/dom` or something?

**Comment by nhunzaker** [comment]
Done. I've also updated the readme with setup instructions.

**Comment by nhunzaker** [comment]
As for inlining the test case descriptions. Every description other than text inputs was pretty low value. Just a quick line that said something like "This fixture tests <textarea />".

However the input README is pretty involved, and talks through some important concepts.

@gaearon did you envision that we would put that information within the text case? Or rather that we'd place instructions for how to use the text fixture along side the text case? Where should we put something like [`fixtures/inputs/README.md`](https://github.com/nhunzaker/react/blob/3d6a63d3c62eefab3f1534e43af7c8451def5e1f/fixtures/dom/src/components/fixtures/text-inputs/README.md)

File: .gitignore [hunk 11]
```diff
@@ -19,6 +19,8 @@ docs/downloads/*.zip
 docs/vendor/bundle
 examples/shared/*.js
 examples/**/bundle.js
+fixtures/dom/public/react-dom.js
+fixtures/dom/public/react.js
 test/the-files-to-test.generated.js
 *.log*
 chrome-user-data
```
Comment (nhunzaker) [inline .gitignore]: Thanks, 👍

**Comment by aweary** [comment]
@nhunzaker I think we could have the READMEs explain high-level concepts and implementation details that are good to know, but not crucial for testing. Then we inline instructions that are just to-the-point directions on how to validate expected behavior. That way we don't include too much detail on the fixtures page, but its still available somewhere.

**Comment by nhunzaker** [comment]
@aweary that would be my preference as well.

**Comment by gaearon** [comment]
Sounds good. Really, the only thing necessary is that it's easy both for the team and the casual contributors to understand what they should test and how it should behave.

**Comment by jquense** [comment]
:wave: just finding this PR. I'd be happy to help out if there is still work to be done :)

File: fixtures/dom/src/components/fixtures/text-inputs/README.md [hunk 12]
```diff
@@ -0,0 +1,65 @@
+# Text Inputs
+
+There are a couple of important concepts to be aware of when working on text
+inputs in React.
+
+## `defaultValue` vs `value`
```
Comment (jquense) [inline fixtures/dom/src/components/fixtures/text-inputs/README.md]: perhaps worth noting that React explicitly supports changing the value of an input imperatively via JavaScript. This isn't an obvious invariant when making changing to input code, but required to probably support autofilling, password managers, and general DOM interop (unfortunately).

File: fixtures/dom/src/components/fixtures/text-inputs/index.js [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+        <input id={id} type={type} />
+      </div>
+    );
+  },
+
+  render() {
+    // https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input
+    let types = [
+      'text', 'email', 'number', 'url', 'tel',
+      'color', 'date', 'datetime-local',
```
Comment (jquense) [inline fixtures/dom/src/components/fixtures/text-inputs/index.js]: AFAIK  `datetime-local` is still supported as `datetime` in some browsers, I'm not sure if that's worth adding or noting tho.

File: fixtures/dom/src/components/fixtures/text-inputs/README.md (hunk 12 above)
Comment (gaearon) [inline fixtures/dom/src/components/fixtures/text-inputs/README.md]: Good catch. We’ll want to add a fixture for this.

**Comment by aweary** [comment]
> 👋 just finding this PR. I'd be happy to help out if there is still work to be done :)

👋 @jquense we still need to write fixtures for all the specific test cases. Right now we just have the "kitchen sink" pages with controlled/uncontrolled variants of a few input types.

I mentioned in https://github.com/facebook/react/pull/8589#issuecomment-270267194 that we might want to write a component that makes it easy to create new fixtures that can be marked as completed (for users validating behavior when making changes)

Any input (heh) you have on scenarios we should be explicitly testing would be amazing! I'll start getting a list together soon and add it to the top comment in this PR.

**Comment by gaearon** [comment]
>Done. I've also updated the readme with setup instructions.
 
Actually I think `fixtures/build` disappeared. This name is probably in gitignore.
You can name it `fixtures/packaging` instead.

**Comment by gaearon** [comment]
After fixing https://github.com/facebook/react/pull/8589#issuecomment-271385313 feel free to merge this when you feel like it. The PR is getting quite large.

**Comment by gaearon** [comment]
In addition to fixing https://github.com/facebook/react/pull/8589#issuecomment-271385313 you'll also need to fix the build fixtures to resolve paths correctly since their relative paths to the build folder are now wrong. You can try to go through the README to see the problem.

**Comment by nhunzaker** [comment]
@gaearon: Sorry for going quiet. Busy day! I should be able to get to this feedback tonight.

**Comment by aweary** [comment]
@nhunzaker I'm working on it right now, I'll get it fixed and then merged 😃

**Comment by gaearon** [comment]
Thanks @aweary!

**Comment by nhunzaker** [comment]
@aweary True friend ❤️

File: fixtures/packaging/webpack/config.js [hunk 14]
```diff
@@ -1,9 +1,11 @@
+var path = require('path');
+
 module.exports = {
   entry: './input',
   output: {
     filename: 'output.js',
   },
   resolve: {
-    root: '../../build/packages',
+    root: path.resolve('../../../build/packages/'),
```
Comment (aweary) [inline fixtures/packaging/webpack/config.js]: @gaearon looks like [webpack requires](https://webpack.github.io/docs/configuration.html#resolve-root) `resolve.root` to be absolute, had to make this change for the webpack packaging fixtures to build

**Comment by aweary** [comment]
Made the required changes, just waiting on CI to approve and will merge after 👍

File: fixtures/packaging/webpack/config.js (hunk 14 above)
Comment (gaearon) [inline fixtures/packaging/webpack/config.js]: Got it, thanks
